AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -109,16 +109,12 @@ - (void)dealloc
 
 #pragma mark - Public methods -- must lock
 
-/// Setter for public image property. It has the side effect to set an internal _imageWasSetExternally that prevents setting an image internally. Setting an image internally should happen with the _setImage: method
+/// Setter for public image property. It has the side effect of setting an internal _imageWasSetExternally that prevents setting an image internally. Setting an image internally should happen with the _setImage: method
 - (void)setImage:(UIImage *)image
 {
   ASDN::MutexLocker l(__instanceLock__);
   
-  _imageWasSetExternally = (image != nil);
-  if (_imageWasSetExternally) {
-    [self _cancelDownloadAndClearImage];
-    _URL = nil;
-  }
+  _imageWasSetExternally = (image != nil && _URL != nil);
```

Comment:
I think this change would make the `ASNetworkImage` back to the old unpredictables behavior. The behavior of the network image node would be based on the order of setting the properties. The behavior would depend on setting the URL before or after setting the image. Without this change it behaved like either an image node or a network image node, now it's kind of a hybrid thing ...